dungeon/src/contrib/item/Item.java

```diff
@@ -132,17 +132,45 @@ public Item(final String displayName, final String description, final Animation
   }
 
   /**
-   * Get the class name of the specific item.
+   * Register an item. This is used to associate the simple name of the class with the class object
+   * in the {@link #REGISTERED_ITEMS} map.
    *
-   * @param id WTF? .
-   * @return The class name of the specific item.
+   * <p>When a new item is created, it should be registered using this method.
+   *
+   * @param clazz The class of the item to register.
+   */
+  public static void registerItem(final Class<? extends Item> clazz) {
+    REGISTERED_ITEMS.put(clazz.getSimpleName(), clazz);
+  }
+
+  /**
+   * Get all registered items.
+   *
+   * <p>This method is used to get all registered items. It returns a map where the keys are the
+   * simple names of the classes (e.g. {@link ItemResourceBerry}), and the values are the
+   * corresponding class objects.
+   *
+   * @return A map of all registered items.
+   */
+  public static Map<String, Class<? extends Item>> registeredItems() {
+    return REGISTERED_ITEMS;
```

Comment:
Brauchst Du das wirklich? 

Ich finde es architektonisch immer etwas schwierig, wenn Klassen ihre Attribute rausreichen ... Damit  fliegt die Kapselung auf, und ich kann "elegant" von außen den inneren Zustand einer Klasse/eines Objekts ändern. Imho deutet das auf Probleme in der Modellierung hin.

Reply:
versteh mich nicht falsch. mir geht es erstmal nur um eine bewusste entscheidung. wenn du das so brauchst, dann ist auch gut ... (dann kann man später ggf. nochmal refactoring machen). wenn nicht, dann besser wieder raus damit :)

Reply:
das meinte ich mit architektur :)

Reply:
egal, darüber kann man mal im gesamtbild nachdenken.